topdown/arithmetic.go

```diff
@@ -123,7 +123,15 @@ func builtinRem(a, b ast.Value) (ast.Value, error) {
 	n2, ok2 := b.(ast.Number)
 
 	if ok1 && ok2 {
-		i, err := arithRem(builtins.NumberToInt(n1), builtins.NumberToInt(n2))
+
+		op1, err1 := builtins.NumberToInt(n1)
+		op2, err2 := builtins.NumberToInt(n2)
+
+		if err1 != nil || err2 != nil {
+			return nil, fmt.Errorf("mod operation supported on integers only")
```

Comment:
It might just be me, but I think this error reads a bit strange.

If you run this in the REPL, this is what you get:

```
> 1.1 % 2
1.1 % 2: eval_internal_error: rem: mod operation supported on integers only
```

```
modulo on floating-point number
```

One thing to note is that because we parse integers using big.Float in all cases, large integer values get represented in scientific notation which big.Int doesn't parse. We should file a separate bug to fix that.

Reply:
Updated the error message.